x-pack/plugins/enterprise_search/public/applications/app_search/components/curations/curation/curation_logic.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { clearFlashMessages, flashAPIErrors } from '../../../../shared/flash_messages';
+import { HttpLogic } from '../../../../shared/http';
+import { KibanaLogic } from '../../../../shared/kibana';
+import { ENGINE_CURATIONS_PATH } from '../../../routes';
+import { EngineLogic, generateEnginePath } from '../../engine';
+
+import { Curation } from '../types';
+
+interface CurationValues {
+  dataLoading: boolean;
+  curation: Curation;
+}
+
+interface CurationActions {
+  loadCuration(): void;
+  onCurationLoad(curation: Curation): { curation: Curation };
+  updateCuration(options?: { queries?: string[] }): { queries?: string[] };
+}
+
+interface CurationProps {
+  curationId: Curation['id'];
+}
+
+export const CurationLogic = kea<MakeLogicType<CurationValues, CurationActions, CurationProps>>({
+  path: ['enterprise_search', 'app_search', 'curation_logic'],
+  actions: () => ({
+    loadCuration: true,
+    onCurationLoad: (curation) => ({ curation }),
+    updateCuration: ({ queries } = {}) => ({ queries }),
+  }),
+  reducers: () => ({
+    dataLoading: [
+      true,
+      {
+        loadCuration: () => true,
+        onCurationLoad: () => false,
+      },
+    ],
+    curation: [
+      {
+        id: '',
+        last_updated: '',
+        queries: [],
+        promoted: [],
+        organic: [],
+        hidden: [],
+      },
+      {
+        onCurationLoad: (_, { curation }) => curation,
+      },
+    ],
+  }),
+  listeners: ({ actions, values, props }) => ({
+    loadCuration: async () => {
+      const { http } = HttpLogic.values;
+      const { engineName } = EngineLogic.values;
+
+      try {
+        const response = await http.get(
+          `/api/app_search/engines/${engineName}/curations/${props.curationId}`
+        );
+        actions.onCurationLoad(response);
+      } catch (e) {
+        const { navigateToUrl } = KibanaLogic.values;
+
+        flashAPIErrors(e, { isQueued: true });
+        navigateToUrl(generateEnginePath(ENGINE_CURATIONS_PATH));
+      }
+    },
+    updateCuration: async ({ queries }, breakpoint) => {
+      const { http } = HttpLogic.values;
+      const { engineName } = EngineLogic.values;
+
+      await breakpoint(100);
```

Comment:
HOLEY MOLEY WHY DID I NOT KNOW ABOUT THIS

Reply:
It's pretty incredible honestly, I legit don't know if I can ever use redux again without Kea 😁 

I also totally forgot to flag/mention this in my earlier comments (brain fart on my part, apologies) but I need this `breakpoint` in order to debounce the following use cases:

- When people spam click a bunch of promote/demote/hide/show actions super fast (high APM gamers :trollface:)
- When clicking the `Reset defaults` buttons, that's actually me calling two simultaneous actions (clearing promoted IDs and clearing hidden IDs), and I need that to be debounced as well and not make two separate API calls

Hope that all makes sense! Also Kea rocks!

Reply:
Another thought - I picked `100` kind of arbitrarily based on what stopped returning errors for me locally. That may vary on actual prod deployments so we should maybe consider doing a more thorough audit of our breakpoint timers and DRYing out some safe standards/levels at some point, as a far future tech debt item. CC @JasonStoltz on that as an FYI